util/ranger/points.go

```diff
@@ -340,6 +345,48 @@ func handleUnsignedCol(ft *types.FieldType, val types.Datum, op string) (types.D
 	return val, op, false
 }
 
+// handleBoundCol handles the case when column meets overflow value.
+// The three returned values are: fixed constant value, fixed operator, and a boolean
+// which indicates whether the range is valid or not.
+func handleBoundCol(ft *types.FieldType, val types.Datum, op string) (types.Datum, string, bool) {
+	isUnsigned := mysql.HasUnsignedFlag(ft.Flag)
+	isNegative := val.Kind() == types.KindInt64 && val.GetInt64() < 0
+	switch ft.Tp {
+	case mysql.TypeTiny, mysql.TypeShort, mysql.TypeInt24, mysql.TypeLong, mysql.TypeLonglong:
+		if !isUnsigned && !isNegative && val.GetUint64() > math.MaxInt64 {
+			switch op {
+			case ast.GT, ast.GE:
+				return val, op, false
+			case ast.NE, ast.LE, ast.LT:
+				op = ast.LE
+				val = types.NewIntDatum(math.MaxInt64)
+			}
+		}
+	case mysql.TypeFloat:
```

Comment:
It's possible to handle the `mysql.TypeDouble`?

Reply:
Maybe we don't need to control the boundary about mysql.TypeDouble. When the value exceeds math.MaxFloat64, an error will be reported when sql parsing.